### PR TITLE
chore: first refactor (add github actions & pnpm)

### PR DESCRIPTION
**Pull request by Aslemammad**
- GitHub actions
- Pnpm (it's far better than the classic npm)

### CODE REVIEW

**Review by NoriSte — COMMENTED** [review]
> Pnpm (it's far better than the classic npm)

What are the main benefits you benefit from on a daily basis?

**Praise**: Thank you ❤️

p.s. I'm using [conventionalcomments](https://conventionalcomments.org/) in my comments.

**Comment by Aslemammad** [comment]
> What are the main benefits you benefit from daily?

Better node_modules structure, and better caching (disk efficient) and it has much fewer lock file issues as opposed to npm. It is also being adopted by mainstream repos (makes it easy for me as a contributor to navigate between repos)

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Test
+on: [push, pull_request]
+
+jobs:
+  short:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version:
+          - 20
```
[the remark this answers is not in the document]
Comment (Aslemammad) [reply .github/workflows/test.yml]: You're correct! It does not need to be tested in different versions.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install pnpm
+        uses: pnpm/action-setup@v2
+
+      - name: Set node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
```
[the remark this answers is not in the document]
Comment (Aslemammad) [reply .github/workflows/test.yml]: Yes, I was going to change this. It's fixed now.

**Review by NoriSte — APPROVED** [review]

**Pull request merged**